packages/utils/src/types.ts

```diff
@@ -2,3 +2,9 @@
 export type Prettify<T> = {
 	[K in keyof T]: T[K];
 } & {};
+
+export type MaybePromise<T> = T | Promise<T>;
+
+export type MaybeFactory<T> = T | (() => T);
+
+export type MaybeThunk<T> = MaybeFactory<MaybePromise<T>>;
```

Comment:
Doesn't seem like too big of a deal but I figured I'd mention, this name didn't click instantly in my head

Reply:
Seemed like the appropriate name for me. It comes from functional programming

> 'thunk' is a function that takes no arguments and, when invoked, performs a potentially expensive operation

Any suggestions that would make this more clear? Would a JSDoc have helped?

Reply:
Ah I just didn't know the term, I suspected it was something meaningful I didn't know about haha

A JSDoc could be cool, but I don't think its truly necessary anyone curious would probably google the term